core/src/raw/oio/list/page_list.rs

```diff
@@ -54,6 +55,11 @@ pub struct PageContext {
     pub done: bool,
     /// token is used by underlying storage services to fetch next page.
     pub token: String,
+    /// key_marker and version_id_marker are used together by underlying storage services to fetch
+    /// next page when object versioning is enabled
+    pub key_marker: String,
```

Comment:
`key_marker` and `version_id_marker` are S3 implementation details that should not be exposed at the `PageList` level. I suggest encoding them in a cheap and storing them in `token`. For example: `format!("{key_marker}|{version_id_marker}")`

Reply:
I want to avoid heavy operations like Base64. I believe `//` or `||` should be safe here since they never appear in version IDs generated by S3. We can use `rsplit_once` to prevent them from showing up in `key_marker`.

Reply:
"since they never appear in version IDs generated by S3", could you let me know where you found this information ? (just out of curiosity).
And could we use `&mut self` in `next_page()` of `trait PageList` ?  this way,  we can manage tokens internally for each service.

Reply:
> "since they never appear in version IDs generated by S3", could you let me know where you found this information ? (just out of curiosity).

Just experiences. There is another way: we can choose an unsafe URL character like a space ` ` or `%`.

Reply:
> Version IDs are Unicode, UTF-8 encoded, **URL-ready**, opaque strings that are no more than 1,024 bytes long

So, Let's use a pace ` `